File: frontend/src/domain/Product/Product.ts


/**
 *  Its @readonly domain @class which contains Product characterizing
 */
class Product {
  readonly id: number;
  readonly name: string;
  readonly fats: number;
  readonly squirrels: number;
  readonly calorificValue: number;
  readonly carbohydrates: number;

  /**
  *  @constructor
  *  @param {number} id
  *  @param {string} name
  *  @param {number} fats
  *  @param {number} squirrels
  *  @param {number} calorificValue
  *  @param {number} carbohydrates
  */
  constructor(id: number, name: string, fats: number, squirrels: number,
      calorificValue: number, carbohydrates: number) {
    this.id = id;
    this.name = name;
    this.fats = fats;
    this.squirrels = squirrels;
    this.calorificValue = calorificValue;
    this.carbohydrates = carbohydrates;
  }
}

export default Product;

File: backend/src/main/java/com/app/hite/core/domain/products/Product.java
package com.app.hite.core.domain.products;



import lombok.Getter;
import lombok.NoArgsConstructor;


import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;


@Getter
@NoArgsConstructor
@Entity
public class Product {


    @Id
    @GeneratedValue
    private Long id;

    private String name;

    private Integer calorificValue;

    private Integer carbohydrates;
    private Integer squirrels;
    private Integer fats;

    public Product(String name, Integer carbohydrates, Integer squirrels, Integer fats, Integer calorificValue) {
        this.name = name;
        this.carbohydrates = carbohydrates;
        this.squirrels = squirrels;
        this.fats = fats;
        this.calorificValue = calorificValue;
    }

}
